modules/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java

```diff
@@ -53,8 +72,135 @@ public ConnectorFuture execute(Context context) {
             url = url.substring(0, url.length() - 1);
             connector.setStringField(0, url);
         }
+
+        HttpConnectionManager connectionManager = HttpConnectionManager.getInstance();
+
+        String scheme;
+        if (url.startsWith("http://")) {
+            scheme = Constants.PROTOCOL_HTTP;
+        } else if (url.startsWith("https://")) {
+            scheme = Constants.PROTOCOL_HTTPS;
+        } else {
+            throw new BallerinaException("Malformed URL: " + url);
+        }
+
+        Map<String, Object> properties =
+                HTTPConnectorUtil.getTransportProperties(connectionManager.getTransportConfig());
+        SenderConfiguration senderConfiguration =
+                HTTPConnectorUtil.getSenderConfiguration(connectionManager.getTransportConfig(), scheme);
+
+        if (connectionManager.isHTTPTraceLoggerEnabled()) {
+            senderConfiguration.setHttpTraceLogEnabled(true);
+        }
+        senderConfiguration.setTlsStoreType(Constants.PKCS_STORE_TYPE);
+
+        BStruct options = (BStruct) connector.getRefField(Constants.OPTIONS_STRUCT_INDEX);
+        if (options != null) {
+            populateSenderConfigurationOptions(senderConfiguration, options);
+            long maxActiveConnections = options.getIntField(Constants.MAX_ACTIVE_CONNECTIONS_STRUCT_INDEX);
+            if (!isInteger(maxActiveConnections)) {
+                throw new BallerinaException("Malformed URL: " + url);
+            }
+            properties.put(Constants.MAX_ACTIVE_CONNECTIONS_PER_POOL, (int) maxActiveConnections);
+        }
+
+        HttpClientConnector httpClientConnector =
+                httpConnectorFactory.createHttpClientConnector(properties, senderConfiguration);
+        connector.setNativeData(Constants.CONNECTOR_NAME, httpClientConnector);
+
         ClientConnectorFuture ballerinaFuture = new ClientConnectorFuture();
         ballerinaFuture.notifySuccess();
+
         return ballerinaFuture;
     }
+
+    private void populateSenderConfigurationOptions(SenderConfiguration senderConfiguration, BStruct options) {
+        //TODO Define default values until we get Anonymous struct (issues #3635)
+        ProxyServerConfiguration proxyServerConfiguration;
+        int followRedirect = 0;
+        int maxRedirectCount = 5;
+        if (options.getRefField(Constants.FOLLOW_REDIRECT_STRUCT_INDEX) != null) {
+            BStruct followRedirects = (BStruct) options.getRefField(Constants.FOLLOW_REDIRECT_STRUCT_INDEX);
+            followRedirect = followRedirects.getBooleanField(Constants.FOLLOW_REDIRECT_INDEX);
+            maxRedirectCount = (int) followRedirects.getIntField(Constants.MAX_REDIRECT_COUNT);
```

Comment:
Shall we rename this constant to MAX_REDIRECT_COUNT_INDEX

Reply:
Ack.